stdlib/universe/narrow_state_tracking.go

```diff
@@ -0,0 +1,171 @@
+package universe
+
+import (
+	"github.com/apache/arrow/go/arrow/memory"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/array"
+	"github.com/influxdata/flux/arrow"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/execute/table"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/values"
+)
+
+func NewNarrowStateTrackingTransformation(t *stateTrackingTransformation, id execute.DatasetID, mem memory.Allocator) (execute.Transformation, execute.Dataset, error) {
+	a := &narrowStateTrackingTransformationAdapter{t: t}
+	nt, d, err := execute.NewNarrowStateTransformation(id, a, mem)
+	if err != nil {
+		return nil, nil, err
+	}
+	return nt, d, nil
+}
+
+type narrowStateTrackingTransformationAdapter struct {
+	t *stateTrackingTransformation
+}
+
+type trackedState struct {
+	start,
+	prevTime values.Time
+
+	countInState,
+	durationInState bool
+
+	count,
+	duration int64
+}
+
+func (a *narrowStateTrackingTransformationAdapter) Process(chunk table.Chunk, state interface{}, d *execute.TransportDataset, mem memory.Allocator) (interface{}, bool, error) {
```

Comment:
Can you logically split this method into different methods based on which part of the algorithm is being implemented? A lot of the older transformations had just massive `Process()` implementations and I'd like the refactored ones to use methods a bit more liberally to be readable. It helps general understandability.